### PR TITLE
feat: Added Option to Share Event

### LINKED ISSUE #975: Option to Share Event in AboutEventFragment
### Feature Request

**Feature Description**

There should be an option in toolbar in About Event Fragment to share the corresponding event as there is in EventBrite app. Clicking on the option will share the necessary details regarding Event on various social media and emailing apps. 

   
**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #975 

Changes: 
1. User can share Event Details from About Event Fragment as in EventBrite.
2. Share Icon color changes according to toolbar's color.
3. An example of info that's shared - 

Event Name

Starts: 23:01, Tue, 10 Apr 2018

Ends at: 03:01, Mon, 30 Apr 2018

Url: https://www.example.com

Screenshots for the change: 
20180527080948-7c011ad592 gif-2-mp4 com

### CODE REVIEW

**Comment by Masquerade0097** [comment]
@iamareebjamal Please Review.

File: app/src/main/java/org/fossasia/openevent/app/core/event/about/AboutEventPresenter.java [hunk 1]
```diff
@@ -116,6 +117,20 @@ private void listenChanges() {
...
+        String doubleLineBreak = "\n\n";
+        if (event.getExternalEventUrl() == null) {
+            data = event.name + doubleLineBreak + "Starts: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.startsAt) +
+                doubleLineBreak + "Ends at: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.endsAt);
+        } else {
+            data = event.name + doubleLineBreak + "Starts: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.startsAt) +
+                doubleLineBreak + "Ends at: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.endsAt) +
+                doubleLineBreak+ "Url: " + event.externalEventUrl;
+        }
+        return data;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/event/about/AboutEventPresenter.java]: Use StringBuilder

File: app/src/main/java/org/fossasia/openevent/app/core/event/about/AboutEventPresenter.java [hunk 2]
```diff
@@ -116,6 +117,20 @@ private void listenChanges() {
...
 
+    public String getShareableInformation() {
+        String data;
+        String doubleLineBreak = "\n\n";
+        if (event.getExternalEventUrl() == null) {
+            data = event.name + doubleLineBreak + "Starts: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.startsAt) +
+                doubleLineBreak + "Ends at: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.endsAt);
+        } else {
+            data = event.name + doubleLineBreak + "Starts: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.startsAt) +
+                doubleLineBreak + "Ends at: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.endsAt) +
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/event/about/AboutEventPresenter.java]: Repetition of formatted date

**Comment by Masquerade0097** [comment]
@iamareebjamal Done !!

File: app/src/main/java/org/fossasia/openevent/app/core/event/about/AboutEventPresenter.java [hunk 3]
```diff
@@ -116,6 +117,18 @@ private void listenChanges() {
             .subscribe(copyrightModelChange -> loadCopyright(false), Logger::logError);
     }
 
+    public String getShareableInformation() {
+        String doubleLineBreak = "\n\n";
+        StringBuilder data = new StringBuilder(event.getName() + doubleLineBreak + "Starts: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.getStartsAt()) +
+            doubleLineBreak + "Ends at: " + DateUtils.formatDateWithDefault(DateUtils.FORMAT_DAY_COMPLETE, event.getEndsAt()));
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/event/about/AboutEventPresenter.java]: This is as good as not using StringBuilder. This is still concatenating

Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/core/event/about/AboutEventPresenter.java]: I've removed concatenation now. Please see if it's fine.

**Pull request merged**